src/SwiftUI/Mac/Color.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+using Foundation;
+using AppKit;
+
+namespace SwiftUI
+{
+	public unsafe partial class Color
+	{
+		#region Constructors
+		public Color (NSColor color)
+		{
+			Data = CreateFromNSColor (color.Handle.ToPointer ());
+		}
+
+		public Color (string name, NSBundle? bundle = null)
+		{
+			Data = CreateFromStringBundle (name, bundle == null ? IntPtr.Zero.ToPointer () : bundle.Handle.ToPointer ());  
+        }
+		#endregion
+
+		#region DllImports
+		// Initialisers
+		[DllImport(SwiftUILib.Path,
+			CallingConvention = CallingConvention.Cdecl,
+			EntryPoint = "$s7SwiftUI5ColorVyACSo7NSColorCcfC")]
+		static extern IntPtr CreateFromNSColor (void* colorPointer);
+
+
+		[DllImport (SwiftUILib.Path,
+			CallingConvention = CallingConvention.Cdecl,
+			EntryPoint = "$s7SwiftUI5ColorV_6bundleACSS_So8NSBundleCSgtcfC")]
+		static extern IntPtr CreateFromStringBundle (
+			[MarshalAs(UnmanagedType.LPWStr)] string str,
+			void* bundlePointer);
```

Comment:
@chkn At the moment passing `IntPtr.Zero` for null, into this param correctly picks up custom asset colour specified.

Reply:
@chkn I've test the above function with both `NSBundle` as `null` and passing in `NSBundle.MainBundle` so that the bundle's pointer is used and both scenarios work.

Using the `Optional.Packed<IntPtr>` looks like it might be the right way to go in the long term. My only concern is, if what we currently have works, maybe we should spend some time to flesh out the `Optional` struct more before integrating it? 

I'll add a reference to this comment, so that both PRs are synced.

Reply:
Thanks for testing this. I bet that Swift treats null as an "[extra inhabitant](https://github.com/apple/swift/blob/master/docs/ABI/TypeLayout.rst#single-payload-enums)" (i.e. a bit pattern that doesn't form a valid value of the type). In that case, there is no special representation for an Optional pointer, and you're absolutely right that we can just leave it as is. Thanks!